File: BoardGame.java
import java.util.*;

/********
 * General Interface for a BoardGame object
 * will let us perform minimiax search and
 * alpha-beta pruning
 ********/

public interface BoardGame {

    public boolean gameOver();

    public boolean hasWon(int color);
    
    public boolean makeMove(Player p, int m);
    
    public ArrayList<Integer> legalMoves(Player p);
    
    public boolean legalMove(Player p, int m);

    public BoardGame clone();

}

File: csharp/Axiom/BoardGame.cs
using System;
using System.Collections.Generic;

namespace Axiom
{
    public interface BoardGame
    {

        public bool GameOver();

        public bool HasWon(Color color);

        public bool MakeMove(Player p, int m);

        public List<int> LegalMoves(Player p);

        public bool LegalMove(Player p, int m);

        public string ShowMoves(Player p);

        public string ShowMove(int m);

        public BoardGame Clone();

        public Color GetTurn();

        public int Heuristic(Color color);

    }
}

